File: code/167.two-sum-ii-input-array-is-sorted.py
#
# @lc app=leetcode id=167 lang=python3
#
# [167] Two Sum II - Input array is sorted
#

# @lc code=start
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        size = len(numbers)
        i = 0
        j = size - 1
        while i < j:
            the_sum = numbers[i] + numbers[j]
            if the_sum == target:
                return [i+1, j+1]
            elif the_sum < target:
                i = i + 1
            else:
                j = j - 1
        return None
# @lc code=end

File: code/167.two-sum-ii-input-array-is-sorted.cpp
/*
 * @lc app=leetcode id=167 lang=cpp
 *
 * [167] Two Sum II - Input array is sorted
 */
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int low = 0;
        int high = numbers.size() - 1;
        int sum = 0;
        while(low < high) {
            sum = numbers[low] + numbers[high];
            if(sum < target) {
                ++low;
            }
            else if(sum > target) {
                --high;
            }
            else {
                break;
            }
        }
        return vector<int>({low+1, high+1});
    }
};

